apps/voting/app/src/script.js

```diff
@@ -356,10 +356,26 @@ function loadVoteSettings() {
     })
 }
 
-async function loadBlockTimestamp(blockNumber) {
-  const { timestamp } = await app.web3Eth('getBlock', blockNumber).toPromise()
-  // Adjust for solidity time (s => ms)
-  return timestamp * 1000
+function loadBlockTimestamp(blockNumber) {
+  // Wrap with retry in case the block is somehow not present
+  return retryEvery(() =>
+    app
+      .web3Eth('getBlock', blockNumber)
+      .toPromise()
+      .then(block => {
+        const { timestamp } = block || {}
+        if (!timestamp) {
+          throw new Error(`Block ${blockNumber} not yet mined`)
+        }
+
+        // Adjust for solidity time (s => ms)
+        return timestamp * 1000
+      })
+      .catch(err => {
+        console.log(`Error fetching block (${blockNumber})`, err)
```

Comment:
I’m thinking that it could be nice to have an API to report errors through aragonAPI one day, so that we could have a journal of all the errors (client and per app) somewhere in the client.

Reply:
Oh yes absolutely; we do detect top-level errors from the script and [(I think) `store()`](https://github.com/aragon/aragon.js/blob/master/packages/aragon-api/src/index.js#L543) but we should definitely ask app developers to wrap their entire app and have a global error handler that pipes messages out to the client.